### DISCUSSION TITLE
Automating RabbitMQ Queue and Exchange Configuration in Aspire

### DISCUSSION THREAD

**Discussion by sharpzilla** [microsoft/aspire#7006] [Q&A] [answered]
I have an application that publishes messages to a RabbitMQ queue and exchange. Currently, after setting up RabbitMQ in Aspire, I have to manually create the queue and exchange to make my application work properly. I would like to automate this process.

I noticed a similar behavior was discussed here: https://github.com/dotnet/aspire/discussions/4584. I tried to use WithBindMount, but after that, my container wouldn't start at all.

Could you please advise on how to preconfigure queues, exchanges, etc., in RabbitMQ?

Here's what I tried:
```
var rabbitmqDefinitionsPath = Path.Combine(Directory.GetCurrentDirectory(), "definitions.json");
var rabbitmqConfigPath = Path.Combine(Directory.GetCurrentDirectory(), "rabbitmq.conf");

var rabbitMq = builder.AddRabbitMQ("messaging")
    .WithImageTag("4.0.5-management")
    .WithManagementPlugin()
    .WithBindMount(rabbitmqConfigPath, "/etc/rabbitmq/rabbitmq.conf")
    .WithBindMount(rabbitmqDefinitionsPath, "/etc/rabbitmq/definitions.json");
```

```
var rabbitMq = builder.AddRabbitMQ("messaging")
    .WithImageTag("4.0.5-management")
    .WithManagementPlugin()
    .WithBindMount("./rabbitmq.conf", "/etc/rabbitmq/rabbitmq.conf")
    .WithBindMount("./definitions.json", "/etc/rabbitmq/definitions.json");
```
definitions.json
```
{
  "vhosts": [
    {
      "name": "/"
    }
  ],
  "queues": [
    {
      "name": "queue1",
      "vhost": "/",
      "durable": true
    }
  ]
}
```
rabbitmq.conf
```
management.load_definitions = /etc/rabbitmq/definitions.json
```

**Comment by davidfowl** [comment, marked as answer]
Do you know what a working configuration looks like without aspire?

**Reply by sharpzilla** [reply]
My bad, I used wrong configuration.
